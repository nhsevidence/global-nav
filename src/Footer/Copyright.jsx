import React, { Component } from "react";

import styles from "./Copyright.module.scss";

export default class Copyright extends Component {
	render() {
		return (
			<div className={styles.copyright}>
				<nav>
					<ul>
						<li>
							<a href="https://www.nice.org.uk/accessibility">Accessibility</a>
						</li>
						<li>
							<a href="https://www.nice.org.uk/freedom-of-information">
								Freedom of information
							</a>
						</li>
						<li>
							<a href="https://www.nice.org.uk/glossary">Glossary</a>
						</li>
						<li>
							<a href="https://www.nice.org.uk/freedom-of-information">
								Terms and conditions
							</a>
						</li>
						<li>
							<a href="https://www.nice.org.uk/privacy-notice">
								Privacy notice
							</a>
						</li>
					</ul>
				</nav>
				<p>
					&copy; NICE {new Date().getFullYear()}. All rights reserved. Subject
					to{" "}
					<a href="https://www.nice.org.uk/terms-and-conditions#notice-of-rights">
						Notice of rights
					</a>
					.
				</p>
			</div>
		);
	}
}
