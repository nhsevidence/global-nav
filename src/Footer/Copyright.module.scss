@import 'settings';

.copyright {
  background: $colour-nice-dark-grey;
}
