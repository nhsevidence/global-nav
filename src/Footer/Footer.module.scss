@import 'footer-settings';

.footer {
  @include links-inverse;
  background: $colour-footer-background;
  box-sizing: border-box;
  color: $colour-footer-text;
  font-family: unquote($nice-font-family-sans);
  margin: rem($spacing-large 0 0 0);

  *,
  *:before,
  *:after {
    box-sizing: inherit;
  }

  a {
    font-weight: normal; // To override Pathways global anchor style
  }
}

.container {
  @include container;
  @include clearfix;
  padding: rem($spacing-medium 0);
}

.logo {
  $logo-width: 80px;
  $logo-aspect-ratio: 512 / 1410;
  display: block;
  height: 0;
  margin: rem(0 0 $spacing-large 0);
  padding: 0;
  padding-bottom: $logo-width * $logo-aspect-ratio; // Aspect ratio of NICE logo SVG
  position: relative;
  width: $logo-width;

  svg {
    display: block;
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: 100%;
  }
}

@include mq($from: md) {
  .logo {
    margin-bottom: rem($spacing-medium);
  }
}
