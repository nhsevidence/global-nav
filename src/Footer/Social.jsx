import React, { Component } from "react";
import FacebookIcon from "@nice-digital/icons/lib/Facebook";
import TwitterIcon from "@nice-digital/icons/lib/Twitter";
import YouTubeIcon from "@nice-digital/icons/lib/YoutubePlay";
//import Instagram from "@nice-digital/icons/lib/Instagram";

import styles from "./Social.module.scss";

export default class Social extends Component {
	render() {
		return (
			<div className={styles.social}>
				<ul>
					<li>
						<a href="https://en-gb.facebook.com/NationalInstituteforHealthandCareExcellence/">
							{typeof SVGRect !== "undefined" && <FacebookIcon />}
							Facebook
						</a>
					</li>
					<li>
						<a href="https://twitter.com/NICEcomms">
							{typeof SVGRect !== "undefined" && <TwitterIcon />}
							Twitter
						</a>
					</li>
					<li>
						<a href="https://www.youtube.com/user/NICEmedia">
							{typeof SVGRect !== "undefined" && <YouTubeIcon />}
							YouTube
						</a>
					</li>
					<li>
						<a href="https://www.instagram.com/nicecomms/">
							{typeof SVGRect !== "undefined" && <YouTubeIcon />}
							Instagram
						</a>
					</li>
				</ul>
			</div>
		);
	}
}
