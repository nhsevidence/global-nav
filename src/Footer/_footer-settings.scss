$colour-footer-background: $colour-nice-black;

$colour-footer-text: $colour-nice-white;
