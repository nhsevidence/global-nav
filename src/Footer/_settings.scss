@import '~@nice-digital/design-system/src/stylesheets/settings/settings';

$colour-footer-background: $colour-nice-black;

$colour-footer-text: $colour-nice-white;
