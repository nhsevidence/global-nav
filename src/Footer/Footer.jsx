import React, { Component } from "react";
import { hot } from "react-hot-loader/root";
import PropTypes from "prop-types";

import NiceLogo from "@nice-digital/icons/lib/Logo";

import Social from "./Social";
import Legal from "./Legal";
import Services from "./Services";
import Pages from "./Pages";
import styles from "./Footer.module.scss";
import TrackedLink from "../TrackedLink";
import { footerClickEventAction } from "../tracker";

export class Footer extends Component {
	render() {
		return (
			<footer className={styles.footer}>
				<div className={styles.container}>
					<TrackedLink
						eventAction={footerClickEventAction}
						eventLabel="Logo"
						href="https://www.nice.org.uk/"
						className={styles.logo}
						aria-label="Go to NICE home page"
					>
						{typeof SVGRect !== "undefined" && <NiceLogo />}
					</TrackedLink>
					<Services service={this.props.service} />
					<Pages service={this.props.service} />
					<Social />
				</div>
				<Legal />
			</footer>
		);
	}
}

Footer.propTypes = {
	service: PropTypes.string
};

export default hot(Footer);
