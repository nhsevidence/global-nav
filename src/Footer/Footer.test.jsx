import React from "react";

import Footer from "./Footer";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Legal", () => {
	it("Renders without crashing", () => {
		const wrapper = shallow(<Footer />);
		expect(wrapper).toHaveLength(1);
	});

	it("Matches snapshot with no SVG support", () => {
		const wrapper = shallow(<Footer service="cks" />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("Matches snapshot with SVG support", () => {
		window.SVGRect = "anything";
		const wrapper = shallow(<Footer service="cks" />);
		expect(toJson(wrapper)).toMatchSnapshot();
		delete window.SVGRect;
	});
});
