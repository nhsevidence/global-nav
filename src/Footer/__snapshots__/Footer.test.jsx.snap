// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legal Matches snapshot with SVG support 1`] = `
<footer
  className="footer"
>
  <div
    className="container"
  >
    <TrackedLink
      aria-label="Go to NICE home page"
      className="logo"
      eventAction="Footer click"
      eventCategory="TopHat and footer"
      eventLabel="Logo"
      href="https://www.nice.org.uk/"
    >
      <SvgLogo />
    </TrackedLink>
    <Services
      service="cks"
    />
    <Pages
      service="cks"
    />
    <Social />
  </div>
  <Legal />
</footer>
`;

exports[`Legal Matches snapshot with no SVG support 1`] = `
<footer
  className="footer"
>
  <div
    className="container"
  >
    <TrackedLink
      aria-label="Go to NICE home page"
      className="logo"
      eventAction="Footer click"
      eventCategory="TopHat and footer"
      eventLabel="Logo"
      href="https://www.nice.org.uk/"
    />
    <Services
      service="cks"
    />
    <Pages
      service="cks"
    />
    <Social />
  </div>
  <Legal />
</footer>
`;
